lib/discover/index.js

```diff
@@ -0,0 +1,21 @@
+var url = require('url')
+
+module.exports.protocols = {
+    "hyperbahn:": require('./hyperbahn').discover,
+    "file:": require('./file').discover,
+    "ip:": require('./ip').discover,
+    // by default fallback on ip discovery
+    default: require('./ip').discover
+}
+
+// find a list of initial nodes to connect to based on the discover string
+module.exports.discover = function (discover, callback) {
```

Comment:
Good catch. I won't do `url` since the url module is already in that name which will cause an overload on that one :)